src/utils/pair_hash.h

```diff
@@ -0,0 +1,31 @@
+#include <math.h>
+
+/* @brief Pair function of Szudzik (2006).
+ * @details Useful to encapsulate two integers as a unique reversible integer hash.
+ * @return hashed integer.
+ * @see https://en.wikipedia.org/wiki/Pairing_function#Cantor_pairing_function
+ */
+static inline int pair_hash(int x, int y) {
```

Comment:
@rafaelcn

Reply:
> hmmm, why did you closed it? lets talk a bit more before merging stuff 🤣 

@rafaelcn

Hahaha sorry, I wanted to move forward with this and forget all that stuff for some days. But I found the idea of pair reversable hashing pretty interesting. With this I used my priority queue that has a key as an integer to implement the Prim's Algorithm, which need to store the edge (u, v) and the distance as priority for a minimum heap.